xtask/src/commands/precommit.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow::Result;
+use clap::Args;
+use xshell::Shell;
+
+use super::common;
+
+#[derive(Args)]
+pub struct Precommit;
+
+impl Precommit {
+    pub fn run(&self, sh: &Shell) -> Result<()> {
+        // Apply rustfmt
+        common::run_fmt(sh)?;
```

Comment:
Can you actually modify files as part of a pre commit hook? Will the changes show up in the commit?